src/Database/LSMTree/Normal.hs

```diff
@@ -335,7 +337,24 @@ withCursor ::
   => TableHandle m k v blob
   -> (Cursor m k v blob -> m a)
   -> m a
-withCursor (Internal.NormalTable th) action = Internal.withCursor th (action . Internal.NormalCursor)
+withCursor (Internal.NormalTable th) action =
+    Internal.withCursor Nothing th (action . Internal.NormalCursor)
+
+{-# SPECIALISE withCursorAtOffset :: SerialiseKey k => k -> TableHandle IO k v blob -> (Cursor IO k v blob -> IO a) -> IO a #-}
+-- | A variant of 'withCursor' that allows initialising the cursor at an offset,
+-- so it will skip all entries with keys less than the provided offset key.
```

Comment:
I think it would be clearer and more precise to say something like:
```suggestion
-- | A variant of 'withCursor' that allows initialising the cursor at an offset
-- within the table such that the first entry the cursor returns will be the
-- one with the lowest key that is greater than or equal to the given key.
-- In other words, it uses an inclusive lower bound.
```